src/utils/swapErrorToUserReadableMessage.ts

```diff
@@ -0,0 +1,52 @@
+// eslint-disable-next-line no-restricted-imports
+import { t } from '@lingui/macro'
+/**
+ * This is hacking out the revert reason from the ethers provider thrown error however it can.
+ * This object seems to be undocumented by ethers.
+ * @param error an error from the ethers provider
+ */
+export function swapErrorToUserReadableMessage(error: any): string {
+  let reason: string | undefined
+
+  if (error.code) {
+    switch (error.code) {
+      case 4001:
+        return t`Transaction rejected`
+    }
+  }
+
+  while (error) {
```

Comment:
the while (error) + reassigning error loop seems kind of strange to me. can this be done by just assigning new variables?

Reply:
I'm not sure if it can, because this is done because there is an unknown number of nested errors here.
fwiw this is legacy code, so it may be better to not touch it unless we have documentation or can repro the errors that it was created to handle.